src/tools/imageSrcset.js

```diff
@@ -0,0 +1,48 @@
+export default class {
+    getSrcset(url, sizes) {
+        // Regex to test size string is of the form 123x123 or 100w
+        const sizeRegex = /(^\d+w$)|(^(\d+?)x(\d+?)$)/;
+        // Regex to test to that srcset descriptor is of the form 1x 1.5x 2x OR 123w
+        const descriptorRegex = /(^\d+w$)|(^([0-9](\.[0-9]+)?)x)$/;
+
+        function isObject(obj) {
```

Comment:
Is there a reason we are nesting it here. Can we move it out into a common lib to be accessible for the entire codebase in the future.